### PR TITLE
Load the Intl.Segmenter and Intl.DurationFormat polyfills only if needed

### PR DESCRIPTION
**Pull request by sandhose**
This removes around 200Kb of the main bundle.

It will still load the polyfill if needed (notably the previous Firefox ESR), but skips on modern browsers.

### CODE REVIEW

File: src/reactions/RaisedHandIndicator.tsx [hunk 1]
```diff
@@ -12,12 +12,11 @@ import {
   useEffect,
   useState,
 } from "react";
-import { DurationFormat } from "@formatjs/intl-durationformat";
 import { useTranslation } from "react-i18next";
 
 import { ReactionIndicator } from "./ReactionIndicator";
 
-const durationFormatter = new DurationFormat(undefined, {
+const durationFormatter = new Intl.DurationFormat(undefined, {
```
Comment (Half-Shot) [inline src/reactions/RaisedHandIndicator.tsx]: This landed in Node 23, so we will need to polyfill the tests.

Comment (sandhose) [reply src/reactions/RaisedHandIndicator.tsx]: Fixed! I also had to embed the translations in tests, as now the initializer waits for i18next to initialise now before continuing

Comment (Half-Shot) [reply src/reactions/RaisedHandIndicator.tsx]: Oh but your polyfill code should actually be doing this, ugh??

Comment (sandhose) [reply src/reactions/RaisedHandIndicator.tsx]: `initializeBeforeReact` isn't necessarily being run in tests

**Review by hughns — APPROVED** [review]

**Pull request merged**